### DISCUSSION TITLE
How to Handle Failed Hunts

### DISCUSSION THREAD

**Discussion by Walishaha** [Velocidex/velociraptor#3044] [Q&A]
Hello Mike and team,

While trying to run multiple hunts in Parallel, we are seeing hunts that fail on some hosts, sometimes this is due to timeout, and sometimes the hosts get offline for any reason. Can you please suggest how to run the hunts on those hosts where it was failed?

My thoughts are to list down the machines where the hunt failed and tag them as uncovered and then copy and run the hunt on those tagged hosts but this is an efficient way I think.

Please advise.

Thank you

**Comment by zachvessey16** [comment]
There's a built-in way to do this without manually tagging/re-running clients.

**Manual (GUI):** Open the failed collection from the Clients tab, copy it to a new collection (you can bump resource limits/timeout there), then use "Add to Hunt" to fold it back into the same hunt.

**Automated (VQL):** This grabs every client in the hunt with a flow in `ERROR` state, re-collects with higher limits, and adds the new flow back into the same hunt automatically:
```
LET NewCollections = SELECT ClientId, FlowId,
    collect_client(client_id=ClientId,
        artifacts=Flow.request.artifacts,
        spec=Flow.request.specs,
        max_bytes=1000000000,
        timeout=600) AS NewCollection
FROM hunt_flows(hunt_id=HuntId)
WHERE Flow.state =~ "ERROR"

SELECT ClientId, NewCollection, hunt_add(
   client_id=ClientId,
   hunt_id=HuntId,
   flow_id=NewCollection.flow_id) AS Hunt
FROM NewCollections
```
Needs commit #2067 or later. Your label-and-rerun idea would also work, but this is the native mechanism and keeps everything under the original HuntID instead of spawning a separate hunt.

Source: [official knowledge base tip](https://docs.velociraptor.app/knowledge_base/tips/retry_hunt/).
